docs: simple basic status, account and product switcher docs (#3255)
twilio-labs/paste: packages/paste-website/src/pages/components/status-menu/index.mdx

  title: 'Status Menu - Components',
  package: '@twilio-paste/status',
  description:

>     'A Status Menu should be used to communicate the status of a process, or connection to an application or service and provide a way to update that status type.',

Reviewer: ```suggestion
    'A Status Menu communicates the status of a process or connectivity to an application or service, and provides a way to update that status type.',
```

> 

Reviewer: todo: The example at the top needs a checked value. Currently, nothing's checked in the list.

![image](https://github.com/twilio-labs/paste/assets/3821840/ad8f0ee3-f49d-45e9-b423-7cdf65aeb962)

## Guidelines

### About Status Menu

> The Status Menu can be used when you want to both display and change the status of a connection or process. It closely follows the [Status Pattern](/patterns/status) and comes with preconfigured options for displaying the status of both processes and connections, using the correct icons for each.

Reviewer: ```suggestion
Use a Status Menu to both display and change the status of a connection or process. It closely follows the [Status Pattern](/patterns/status) and comes with preconfigured options for displaying the status of both processes and connections, using the correct icons for each.
```

### Accessibility

> The Status Menu implements a Menu and supports all the same keyboard navigation that the [Menu](/components/menu) component does.

Reviewer: ```suggestion
Status Menu implements a Menu and supports all the same keyboard navigation that the [Menu](/components/menu) component does.
```

## Examples

### Process menu

> Use the Process related variants of Status Menu to display and interact with the status of a particular process.

Reviewer: ```suggestion
Use the process-related variants of Status Menu to display and interact with the status of a particular process.
```

  {processStatusMenuExample}
</LivePreview>

### Connectivity Menu

> Use the Connectivity related variants of Status Menu to display and interact with the status of a particular connection.

Reviewer: ```suggestion
Use the connectivity-related variants of Status Menu to display and interact with the status of a particular connection.
```

## Composition notes

> The Status Menu consists of `StatusMenu`, `StatusMenuBadge`, `StatusMenuItem`, `StatusMenuItemRadio`, `StatusMenuItemCheckbox`, and `useStatusMenuState`. These are all name sakes of their [Menu](/components/menu) equivilants. It also comes with one additional component called `StatusMenuItemChild`.

Reviewer: ```suggestion
Status Menu consists of `StatusMenu`, `StatusMenuBadge`, `StatusMenuItem`, `StatusMenuItemRadio`, `StatusMenuItemCheckbox`, and `useStatusMenuState`. These are all namesakes of their [Menu](/components/menu) equivalents and follow the same API. It also comes with one additional component called `StatusMenuItemChild`.
```

### Status Menu Item Child

> The `StatusMenuItemChild` is where the status pattern is implemented. It takes a `variant` prop that be set to a value that directly relates to one of the connection or process status types. It pairs the correct icon, with the color implementation and text of your choice.

Reviewer: ```suggestion
`StatusMenuItemChild` is where the status pattern is implemented. It takes a `variant` prop that is set to a value that directly relates to one of the connectivity or process status types. It pairs the correct icon with its color implementation for the status. Check out the [full list of variant options in the props table](#statusmenuitemchild).
```
